src/main/java/seedu/weme/commons/util/FileUtil.java

```diff
@@ -1,28 +1,123 @@
 import java.util.Optional;
 import java.util.UUID;
 
+import seedu.weme.model.DirectoryPath;
+
+
 /**
  * Writes and reads files
  */
 public class FileUtil {
 
     public static final String MESSAGE_READ_FILE_FAILURE = "Error encountered while reading the file %s";
     public static final String MESSAGE_COPY_FAILURE_SOURCE_DOES_NOT_EXIST = "Copy failed: source file does not exist";
+    public static final String MESSAGE_COPY_FAILURE_INVALID_DIRECTORY = "Copy failed: Invalid Directory Given";
+    public static final String MESSAGE_COPY_FAILURE_INTERNAL_ERROR = "Copy failed: Internal Error Encountered";
+    private static final int INITIAL_FILE_LABEL = 1;
+
 
     private static final String CHARSET = "UTF-8";
 
-    public static boolean isFileExists(Path file) {
+    public static boolean fileExists(Path file) {
         return Files.exists(file) && Files.isRegularFile(file);
     }
 
+    /**
+     * Copies a list of files given by their path to a given directory.
+     *
+     * @param pathList list of paths containing files.
+     * @param folderPath directory path to copy to.
+     * @throws IOException error encountered while copying.
+     */
+    public static void copyFiles(List<Path> pathList, Path folderPath) throws IOException {
+        try {
+            int fileLabel = INITIAL_FILE_LABEL;
+            for (Path path : pathList) {
+                String newFilePath;
+                do {
+                    newFilePath = buildFilePath(folderPath, fileLabel, path);
+                    fileLabel = incrementFileLabel(fileLabel);
+                } while (fileExists(Paths.get(newFilePath)));
+
+                if (isValidPath(newFilePath)) {
+                    copy(path, Paths.get(newFilePath));
+                } else {
+                    throw new IOException(MESSAGE_COPY_FAILURE_INVALID_DIRECTORY);
+                }
+            }
+        } catch (IOException e) {
+            throw new IOException(MESSAGE_COPY_FAILURE_INTERNAL_ERROR);
+        }
+    }
+
+    /**
+     * Returns a list of valid Image files found in the given directory path.
+     *
+     * @param directoryPath Path containing memes to load.
+     * @return List of loadable paths.
+     */
+    public static List<Path> loadImagePath(DirectoryPath directoryPath) throws IOException {
+        List<Path> imageList = new ArrayList<>();
+        recursiveLoad(imageList, directoryPath);
+        return imageList;
+    }
+
+    /**
+     * Recursively loads all valid image files in a given directory.
+     *
+     * @param imageList
+     * @param directoryPath
+     * @throws IOException
+     */
+    private static void recursiveLoad(List<Path> imageList, DirectoryPath directoryPath) throws IOException {
+        final File folder = toFile(directoryPath);
+        for (final File fileEntry : folder.listFiles()) {
+            if (fileEntry.isDirectory()) {
+                recursiveLoad(imageList, new DirectoryPath(fileEntry.getPath())); // recursive call
```

Comment:
could we just overload the constructor for directorypath and directly pass `fileEntry.toPath()` into directorypath?

Reply:
The code won't necessarily be simpler, another method to validate given path input (currently already have isValidDirectoryPath(String) has to be defined.